invenio_records_permissions/api.py

```diff
@@ -8,17 +8,22 @@
 
 """Invenio Records Permissions API."""
 
-from __future__ import absolute_import, print_function
-
 from elasticsearch_dsl.query import Q
 from flask import current_app
 from invenio_search.api import DefaultFilter, RecordsSearch
 
+from .factories import record_read_permission_factory
+
 
 def rdm_records_filter():
     """Records filter."""
     # TODO: Implement with new permissions metadata
-    perm_factory = current_app.config['RECORDS_REST_ENDPOINTS']['recid']['read_permission_factory_imp']()  # noqa
+    try:
+        perm_factory = current_app.config["RECORDS_REST_ENDPOINTS"]["recid"][
+            "read_permission_factory_imp"
+        ]()  # noqa
+    except KeyError:
+        perm_factory = record_read_permission_factory
```

Comment:
Ah I see. You mean the key is not always defined and yet we want a default. Just to remove the nesting, the different control flows and the `noqa` (minor - I know this might not be merged ;) ) 
```suggestion
    perm_factory = (
        current_app.config.get("RECORDS_REST_ENDPOINTS", {})
        .get("recid", {})
        .get("read_permission_factory_imp", record_read_permission_factory)
    ```

Reply:
Your solution is cleaner. But my main point was to show this shouldn't happen (I know git blame points to me 😓 ).

Do you agree that it should be moved (and potentially refactored)  to `invenio-rdm-records`? It is **RDM** records specific search filtering... Something more generic should be placed here (e.g. simply `record_read_permission_factory`)

Reply:
Oh yes - I believe the TODO on line 36 was written by me ;)